---
title: "DESeq2 and GO analysis in R"
output:
    html_document:
      theme: united
      toc: true
      toc_depth: 3
      toc_float: true
      number_section: true
---

```{css, echo=FALSE}
.header-section-number::after {
  content: ".";
}
```

Список пакетов, используемых в работе:
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = F)

```

```{r echo = FALSE, message=FALSE, warning=FALSE}
# The following packages are needed for proper compilation
if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
if (!require(DESeq2)) BiocManager::install("DESeq2")
if (!require(EnhancedVolcano)) BiocManager::install("EnhancedVolcano")
if (!require(dplyr)) install.packages("dplyr")
if (!require(RColorBrewer)) install.packages("RColorBrewer")
if (!require(gplots)) install.packages("gplots")
if (!require(DOSE)) install.packages('DOSE')
if (!require("clusterProfiler")) BiocManager::install("clusterProfiler", version = "3.9")
if (!require(pathview)) BiocManager::install("pathview")
if (!require(enrichplot)) BiocManager::install("enrichplot")
```


```{r echo = T, warning=FALSE, message=FALSE, eval=TRUE}
library(DESeq2)
library(dplyr)
library(RColorBrewer)
library(gplots)
library(EnhancedVolcano)
library(DT)
library(clusterProfiler)
library(enrichplot)
library(ggplot2)

```

# DESeq2
## Data pre-processing
Read the file with featureCounts results.
```{r}
countdata <- read.table("feature_counts.txt", header=TRUE, row.names=1)
```

Pre-processing of dataset for DESeq2 analysis.
```{r}
# Remove first five columns (chromosome, start, end, strand, length)
countdata <- countdata[ ,6:ncol(countdata)]

# Rename columns according to your samples.
colnames(countdata) <- c("Species_A1", "Species_A2", "Species_B1","Species_B2")

# Convert to matrix
countdata <- as.matrix(countdata)
```

## Condition assignment
DESeq needs a condition, which includes "control" and "experiment". You may assign whichever sample type as control (e.x. we assigned `Species_A` as control), and the rest one as experiment (e.x. we assigned `Species_B` as experiment). So that DESeq2 compares gene expression of experiment vs control model.
In the present dataset we had 2 biological replicates of `Species_A` and `Species_B` , so that we created the following condition:
```{r}
(condition <- factor(c(rep("ctl", 2), rep("exp", 2))))

(coldata <- data.frame(row.names = colnames(countdata), condition))

dds <- DESeqDataSetFromMatrix(countData = countdata, colData = coldata, design = ~ condition)
```

## Running DESeq2
The following commands generate new dataset
```{r}
# Run DESeq2
dds <- DESeq(dds)

# Format the results
res <- results(dds)
head(res)
```
## Volcano plot
Now it is time to visualization. One of the best way to visualize differential gene expression is Volcano plot.
```{r}
EnhancedVolcano(res,
                lab = rownames(res),
                x = 'log2FoldChange',
                y = 'pvalue',
                pCutoff = 10e-12,
                xlab = 'Species B')
```

Results may be sorted and saved.
```{r}
# Sort the results data frame by the padj and foldChange columns.
sorted <- res[with(res, order(padj, -log2FoldChange)), ]
head(sorted)
# Turn it into a dataframe to have proper column names.
sorted.df <- data.frame("id"=rownames(sorted),sorted)

# Write the table out.
write.table(sorted.df, file="results.txt", sep="\t", col.names=NA, quote=FALSE)
```


## Data normalization and heatmap 
Another way to visualize differential gene expression is a heatmap. 
Firstly, data must be normalized.

```{r}
# Get normalized counts
nc <- counts(dds,normalized=TRUE)

# Turn it into a dataframe to have proper column names.
data <- data.frame("id"=rownames(nc),nc)

gene <- data[,1]
vals <- as.matrix(data[,2:ncol(data)])

# Adds a little noise to each element to avoid the clusting function failing on zero variance datalines.
vals = jitter(vals, factor = 1, amount=0.00001)

# Calculate zscore
score = NULL
for (i in 1:nrow(vals)) {
  row = vals[i,]
  zscore = (row-mean(row))/sd(row)
  score = rbind(score,zscore)
}

row.names(score) <- gene
zscore <- score

# Generate heatmap
mat <- as.matrix(zscore)
```

Draw a heatmap. Note that if you have big data it may takes up to 20 minutes (depends on your computer).

```{r}
pdf('output.pdf')

colors <- colorRampPalette(c("green", "black", "red"),space = "rgb")(256)
heatmap.2(mat, col = pal, density.info = "none",trace = "none", margins = c(14,14), lhei = c(1,5))

invisible(dev.off())
```

# GO analysis
GO analysis allows to group differentially expressed genes by their molecular function, biological process or cellular component.

Firstly, reference genome annotation must be downloaded from http://bioconductor.org/packages/release/BiocViews.html#___OrgDb 
We investigated the differential gene expression in voles, so we have decided to choose Rat annotation as it is the closest relative to vole among all organisms presented in abovementioned database.
```{r}
organism = "org.Rn.eg.db"
#BiocManager::install(organism, character.only = TRUE)
library(organism, character.only = TRUE)
```

On the next step gene key type must be checked. It is very important as clusterProfiler cannot recognize gene key type automatically. Therefore right gene key type must be denoted manually.
Here is an example of gene key type conversion from SYMBOL to ENSEMBL.
```{r}
df <- sorted.df
# Convert gene key type
gene_list <- as.character(df$id)
gene_ensemble <- select(org.Rn.eg.db,
       keys = gene_list,
       columns = c('ENSEMBL'),
       keytype = 'SYMBOL')

gene_ensemble_uniq <- distinct(gene_ensemble, SYMBOL, .keep_all = TRUE) 

df_n <- cbind(df, gene_ensemble_uniq)
```


```{r}
# Log2 fold change is needed
original_gene_list <- df$log2FoldChange

# Name the vector
names(original_gene_list) <- df_n$ENSEMBL

# Omit any NA values 
gene_list <- na.omit(original_gene_list)

# Sort the list in decreasing order (required for clusterProfiler)
gene_list = sort(gene_list, decreasing = TRUE)

# View the acceptible key gene types
keytypes(org.Rn.eg.db)

gse <- gseGO(geneList=gene_list, 
             ont ="BP", 
             keyType = "ENSEMBL", 
             nPerm = 10000, 
             minGSSize = 3, 
             maxGSSize = 1000,
             pvalueCutoff = 0.001,
             verbose = TRUE, 
             OrgDb = organism, 
             pAdjustMethod = "none")

dotplot(gse, showCategory=10, split=".sign") + facet_grid(.~.sign)
```




